internal/controllers/app_controller.go

```diff
@@ -828,6 +847,34 @@ func (r *AppReconciler) deleteChart(ctx context.Context, app *ketchv1.App) error
 
 }
 
+//UpdateAppNamespaceLabelsForIngress updates an app's namespace labels to account for different ingresses.
+// we rely on istio automatic sidecar injection
+// https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#automatic-sidecar-injection
+func (r *AppReconciler) UpdateAppNamespaceLabelsForIngress(ctx context.Context, app *ketchv1.App) error {
+	var framework ketchv1.Framework
+	err := r.Client.Get(ctx, types.NamespacedName{Name: app.Spec.Framework, Namespace: app.ObjectMeta.Namespace}, &framework)
+	if err != nil {
+		return err
+	}
+	namespacedName := framework.Spec.NamespaceName
+	if app.Spec.Namespace != "" { // override framework.spec.namespace if app.spec.namespace is set
+		namespacedName = app.Spec.Namespace
+	}
+	var namespace v1.Namespace
+	err = r.Get(ctx, types.NamespacedName{Name: namespacedName}, &namespace)
+	if err != nil {
+		return err
+	}
+	namespace.Labels["istio-injection"] = "enabled"
+	if framework.Spec.IngressController.IngressType != ketchv1.IstioIngressControllerType {
+		delete(namespace.Labels, "istio-injection")
+		app.AddLabel(map[string]string{"sidecar.istio.io/inject": "false"}, ketchv1.Target{Kind: "Pod", APIVersion: "v1"})
```

Comment:
with `"sidecar.istio.io/inject": "false"` label should we just keep a namespace as it is? 
meaning we really don't need to care about a namespace's labels.

Reply:
Sounds good. I removed all the namespace label changes.